### PR TITLE
Text analytics error model

### LINKED ISSUE #8864: Design exception model consistent with Cognitive Services error handling patterns
- Understand the Cognitive Services error library and patterns
Conceptual: https://dev.azure.com/msazure/Cognitive%20Services/_wiki/wikis/Cognitive%20Services.wiki/15797/Errors-Long-running-operations
Library: https://dev.azure.com/msazure/Cognitive%20Services/_packaging?_a=package&feed=CognitiveServices&package=Microsoft.CloudAI.Containers.Common.Errors&version=1.1.10490001&protocolType=NuGet
Code: https://dev.azure.com/msazure/Cognitive%20Services/_git/Platform-Containers-Common?path=%2Fsrc%2FCommon.Errors&version=GBmaster

Decide how to expose these errors to users.

Goal: Developers can tell what's going wrong easily.

### PR DESCRIPTION
**Pull request by maririos**
Fixes: #8864 
It also makes single method return types atomic (raw information)

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs [hunk 1]
```diff
@@ -103,8 +101,7 @@ public async Task ExtractKeyPhrasesTest()
             TextAnalyticsClient client = GetClient();
             string input = "My cat might need to see a veterinarian.";
 
-            ExtractKeyPhrasesResult result = await client.ExtractKeyPhrasesAsync(input);
-            IReadOnlyCollection<string> keyPhrases = result.KeyPhrases;
+            IList<string> keyPhrases = (IList<string>)await client.ExtractKeyPhrasesAsync(input);
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs]: @pakrym what is the correct way to call this here?  `Response<IList<string>> keyPhrases = await client.ExtractKeyPhrasesAsync(input);` ?

**Comment by maririos** [comment]
@pakrym I just saw https://github.com/Azure/azure-sdk-for-net/pull/9782
looks like it is going to affect my PR too (like it did for Storage), right?
Is this change required for the release for next week?

**Comment by pakrym** [comment]
> @pakrym I just saw #9782
> looks like it is going to affect my PR too (like it did for Storage), right?

I would, but I'm glad I've caught it and we don't have another copy of the formatting code around. Let's chat and I'll help you migrate to the new system.

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs (hunk 1 above)
Comment (pakrym) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs]: I prefer the `Response<IList<string>> keyPhrases = await client.ExtractKeyPhrasesAsync(input);` options.

**Comment by maririos** [comment]
closing draft while I adopt changes from #9782

**Pull request closed without merging**